lambda_src/drivers/destination_s3.py

```diff
@@ -101,13 +100,19 @@ def write(
     row_index: int,
 ) -> Dict[Text, Any]:
     bucket, prefix = parse_destination_uri(destination)
-    encoded_datum = (
-        '\n'.join(json.dumps(d) for d in datum)
-        if isinstance(datum, list)
-        else json.dumps(datum, default=str)
-    )
+    if isinstance(datum, bytes):
+        encoded_datum = datum
+        prefixed_filename = hash_format(
+            prefix, encoded_datum, sha2=lambda x: sha256(x).hexdigest()
```

Comment:
instead of taking it explicitly, when defining a lambda you can "enclose" variables in the scope to create a [closue](https://www.geeksforgeeks.org/python-closures/), i.e. you can do `lambda: sha256(encoded_datum).hexdigest()` here and it will do what you'd expect